src/TraceEvent/TraceEventNativeMethods.cs

```diff
@@ -499,33 +588,93 @@ internal unsafe struct EVENT_FILTER_EVENT_ID
 
         [DllImport("advapi32.dll", CharSet = CharSet.Unicode)]
         internal static extern int StartTraceW(
-            [Out] out UInt64 sessionHandle,
+            [Out] out TRACEHANDLE sessionHandle,
             [In] string sessionName,
             EVENT_TRACE_PROPERTIES* properties);
 
+        internal static int StartTrace(
+            out SafeTraceHandle sessionHandle,
+            string sessionName,
+            EVENT_TRACE_PROPERTIES* properties)
+        {
+            int dwErr = StartTraceW(out TRACEHANDLE dangerousHandle, sessionName, properties);
+            if (dwErr == 0)
+            {
+                sessionHandle = new SafeTraceHandle(dangerousHandle);
+                return dwErr;
+            }
+
+            if (IsValidTraceHandle(dangerousHandle))
+            {
+                CloseTrace(dangerousHandle);
+            }
+
+            sessionHandle = null;
+            return dwErr;
+        }
+
         [DllImport("advapi32.dll", CharSet = CharSet.Unicode)]
         internal static extern int EnableTrace(
             [In] uint enable,
             [In] int enableFlag,
             [In] int enableLevel,
             [In] ref Guid controlGuid,
-            [In] ulong sessionHandle);
+            [In] TRACEHANDLE sessionHandle);
+
+        internal static int EnableTrace(
+            uint enable,
+            int enableFlag,
+            int enableLevel,
+            Guid controlGuid,
+            SafeTraceHandle sessionHandle)
+        {
+            return EnableTrace(
+                enable,
+                enableFlag,
+                enableLevel,
+                ref controlGuid,
+                sessionHandle.DangerousGetHandle());
+        }
 
         [DllImport("advapi32.dll", CharSet = CharSet.Unicode)]
         internal static extern int EnableTraceEx(
             [In] ref Guid ProviderId,
             [In] Guid* SourceId,
-            [In] ulong TraceHandle,
+            [In] TRACEHANDLE TraceHandle,
             [In] int IsEnabled,
             [In] byte Level,
             [In] ulong MatchAnyKeyword,
             [In] ulong MatchAllKeyword,
             [In] uint EnableProperty,
             [In] EVENT_FILTER_DESCRIPTOR* filterData);
 
+        internal static int EnableTraceEx(
+            Guid ProviderId,
```

Comment:
Should this be `ref Guid ProviderId`?

Reply:
Since it's not a DllImport, I took the opportunity to remove the ref. It's a wrinkle that the GUIDs are passed byref in the P/Invoke definition, meaning you have to use "ref providerId" at the call-site. I ironed out the wrinkle for when you're using the SafeHandle overload.
I wonder if `in` would work.

Reply:
`in` works as a replacement for `ref` where it's an input-only value. Will use that.

Reply:
Use `in` instead of `ref`